botocore/docs/bcdoc/restdoc.py

```diff
@@ -205,7 +221,25 @@ def flush_structure(self):
                     self.style.link_target_definition(refname, link)
         value = self.getvalue()
         for name, section in self._structure.items():
-            value += section.flush_structure()
+            # Checks is the AWS API Documentation link has been generated.
+            # If it has been generated, it gets passed as a the doc_link parameter.
+            if b'`AWS API Documentation <' in value:
+                start = value.find(b'`AWS API Documentation <')
+                end = value.find(b'>`_', start) + 3
+                docs_link = value[start:end] + b'\n\n'
+            value += section.flush_structure(docs_link)
```

Comment:
What is the scope of `_structure` here, is it only a single operation or the whole doc model? The case I'd be worried about is this code sets `doc_link` as soon as `b'`AWS API Documentation <'` is encountered. That means every future item in the for loop will always have this value set, right? Unless we override it again, or set it back to None, are we potentially writing erroneous links?

Reply:
Generating some test samples, it doesn't appear this is happening, so the scoping for `_structure` may be limited enough to prevent this.

Reply:
Yeah agree with Nate here. If there's an easy way to inject the link shortly after you replace the text with `LARGE_SECTION_MESSAGE` that would be ideal

Reply:
So whats happening here is we're recursively going through each service's sub-structures in a DFS style. An example path for quicksight's `cancel_ingestion` is `['quicksight', 'client-api', 'methods', 'cancel_ingestion']`. The `cancel_ingestion` section will then have the following available sub-sections sections `method-intro`, `request-example`, `request-params`, `return`, & `exceptions`. The documentation link is found in `method-intro` and then passed into the other sub-sections (happens in the for loop). Nate is right that a documentation link can wrongfully be passed into a subsequent method's traversal. I'll add a condition that resets the `docs_link` to `None` when `len(self.path) == 4` since this indicates a new method.